iac-common/src/test/java/org/sonar/iac/common/checks/PrivilegeEscalationVectorTest.java

```diff
@@ -74,4 +83,96 @@ void all_vectors_have_a_non_empty_name_and_permissions_list() {
+      ":CreatePolicyVersion")));
+    assertThat(actualVector).containsSame(PrivilegeEscalationVector.CREATE_POLICY_VERSION);
+  }
+
+  @Test
+  void getStatementEscalationVectorNoAllowEffect() {
+    Policy.Statement statement = statement(Map.of());
+
+    Optional<PrivilegeEscalationVector> actualVector = getStatementEscalationVector(statement, List.of(text("iam" +
+      ":CreatePolicyVersion")));
+    assertThat(actualVector).isEmpty();
+  }
+
+  @Test
+  void getStatementEscalationVectorWithResourceNotSensitive() {
+    Policy.Statement statement = statement(Map.of("Effect", "Allow",
+      "Action", "action",
+      "Resource", "NotSensitive"));
+
+    Optional<PrivilegeEscalationVector> actualVector = getStatementEscalationVector(statement, List.of(text("iam" +
+      ":CreatePolicyVersion")));
+    assertThat(actualVector).isEmpty();
+  }
+
+  @Test
+  void getStatementEscalationVectorWithResourceInPattern() {
+    Policy.Statement statement = statement(Map.of("Effect", "Allow",
+      "Action", "action",
+      "Resource", "arn:foo:bar:baz:bax:user/*"));
+
+    Optional<PrivilegeEscalationVector> actualVector = getStatementEscalationVector(statement, List.of(text("iam" +
+      ":CreatePolicyVersion")));
+    assertThat(actualVector).containsSame(PrivilegeEscalationVector.CREATE_POLICY_VERSION);
+  }
+
+  @Test
+  void getStatementEscalationVectorConditionExists() {
+    Policy.Statement statement = statement(Map.of("Effect", "Allow",
+      "Action", "action",
+      "Resource", "*",
+      "Condition", "condition"));
+
+    Optional<PrivilegeEscalationVector> actualVector = getStatementEscalationVector(statement, List.of(text("iam" +
+      ":CreatePolicyVersion")));
+    assertThat(actualVector).isEmpty();
+  }
+
+  @Test
+  void getStatementEscalationVectorPrincipalExists() {
+    Policy.Statement statement = statement(Map.of("Effect", "Allow",
+      "Action", "action",
+      "Resource", "*",
+      "Principal", "principal"));
+
+    Optional<PrivilegeEscalationVector> actualVector = getStatementEscalationVector(statement, List.of(text("iam" +
+      ":CreatePolicyVersion")));
+    assertThat(actualVector).isEmpty();
+  }
+
+  @Test
+  void getStatementEscalationVectorActionNotPresent() {
+    Policy.Statement statement = statement(Map.of("Effect", "Allow",
+      "Resource", "*"));
+
+    Optional<PrivilegeEscalationVector> actualVector = getStatementEscalationVector(statement, List.of(text("iam" +
+      ":CreatePolicyVersion")));
+    assertThat(actualVector).isEmpty();
+  }
+
+  private static Policy.Statement statement(Map<String, String> properties) {
+    Tree[] trees = properties.entrySet().stream()
+      .map(e -> new PolicyTest.TestPropertyTree(e.getKey(), text(e.getValue())))
+      .toArray(Tree[]::new);
+    Tree statementTree = new PolicyTest.TestTree(trees);
+    Tree tree = new PolicyTest.TestTree(statementTree);
+    Policy policy = new Policy(tree, Tree::children);
+    return policy.statement().get(0);
```

Comment:
```suggestion
  private static Policy.Statement statement(Map<String, String> properties) {
    PropertyTree[] propertyTrees = properties.entrySet().stream()
      .map(e -> attribute(e.getKey(), text(e.getValue())))
      .toArray(PropertyTree[]::new);
    TestPropertiesTree statement = properties(propertyTrees);
    return new Policy(null, tree -> List.of(statement)).statement().get(0);
```
Seeing this again, and the reported Code Smell, I believe we can simplify it as the suggestion. I found that it is done in a simpler way in `policy.PolicyValidatorTest` that uses common test helpers.